tsdb/chunks/chunks.go

```diff
@@ -252,34 +257,58 @@ func MergeOverlappingChunks(chks []Meta) ([]Meta, error) {
 			last++
 			continue
 		}
-		nc := &newChks[last]
-		if c.MaxTime > nc.MaxTime {
-			nc.MaxTime = c.MaxTime
-		}
-		chk, err := MergeChunks(nc.Chunk, c.Chunk)
+		dividedChks, err := MergeChunks(newChks[last].Chunk, c.Chunk, MaxSamplesInChunk)
```

Comment:
I think we are fine, but worth to add some tricky cases of 3 chunks overlapping 1:1 with more than 120 samples to our tests?

To check if we are still working well with the comment from line 251.
```
// We need to check only the last chunk in newChks.
// Reason: (1) newChks[last-1].MaxTime < newChks[last].MinTime (non overlapping)
//         (2) As chks are sorted w.r.t. MinTime, newChks[last].MinTime < c.MinTime.
// So never overlaps with newChks[last-1] or anything before that.
```

Reply:
Each chunk from upstream will not have more than 120 samples right?
I'm going to assume you meant 1:1 overlapping for a number of samples and then a non-overlapping extension (which would have to be broken because it will be greater than 120). If I'm wrong then please let me know. :)

Reply:
It can have more than 120 as you just create code that splits. Before that people were merging chunks to XXXXX samples without any logic, right? ;p

Reply:
> I'm going to assume you meant 1:1 overlapping for a number of samples and then a non-overlapping extension (which would have to be broken because it will be greater than 120). 

I mean 3x120 samples block for times [0  10] [0 10] [0 10] from 3 different blocks. (:

Reply:
Done. Please have a look when you're available :)